internal/parser/references.go

```diff
@@ -4,28 +4,47 @@ import (
 	"strings"
 
 	"github.com/microsoft/typescript-go/internal/ast"
+	"github.com/microsoft/typescript-go/internal/checker"
 	"github.com/microsoft/typescript-go/internal/core"
 	"github.com/microsoft/typescript-go/internal/tspath"
 )
 
-func collectExternalModuleReferences(file *ast.SourceFile) {
+func (p *Parser) createSyntheticImport(text string, file *ast.SourceFile) *ast.Node {
+	externalHelpersModuleReference := p.factory.NewStringLiteral(text)
+	importDecl := p.factory.NewImportDeclaration(nil /*modifiers*/, nil /*importClause*/, externalHelpersModuleReference, nil)
 	// !!!
+	// addInternalEmitFlags(importDecl, InternalEmitFlags.NeverApplyImportHelper);
+	externalHelpersModuleReference.Parent = importDecl
+	importDecl.Parent = file.AsNode()
+	// explicitly unset the synthesized flag on these declarations so the checker API will answer questions about them
+	// (which is required to build the dependency graph for incremental emit)
+	externalHelpersModuleReference.Flags &= ^ast.NodeFlagsSynthesized
+	importDecl.Flags &= ^ast.NodeFlagsSynthesized
+	return externalHelpersModuleReference
+}
+
+func (p *Parser) collectExternalModuleReferences(file *ast.SourceFile) {
+	isJavaScriptFile := ast.IsInJSFile(file.AsNode())
+	isExternalModuleFile := ast.IsExternalModule(file)
+	// !!!
+	var options *core.CompilerOptions
```

Comment:
stuck here w/ no clear way to access `options`. the call chain is:

- `ParseSourceFile`
  - `parseSourceFileWorker`
    - `collectExternalModuleReferences`